src/test/validation_chainstatemanager_tests.cpp

```diff
@@ -541,7 +541,7 @@ BOOST_FIXTURE_TEST_CASE(chainstatemanager_snapshot_init, SnapshotTestSetup)
     {
         LOCK2(::cs_main, bg_chainstate.MempoolMutex());
         BOOST_CHECK(bg_chainstate.DisconnectTip(unused_state, &unused_pool));
-        unused_pool.clear();  // to avoid queuedTx assertion errors on teardown
+        unused_pool.take();  // to avoid queuedTx assertion errors on teardown
```

Comment:
Why prefer `take()` here? Seems like that's relying on the side-effect, as opposed to `clear()` which does the thing we want.

Reply:
It just seemed like slightly more realistic usage but that doesn't seem to be a concern here, so changed to `clear()` now.